src/clients/safe_client_gateway.py

```diff
@@ -16,18 +16,27 @@ def setup_session() -> requests.Session:
     session.mount("https://", adapter)
     return session
 
+def valid_url(url) -> bool:
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc])
+    except ValueError:
+        return False
```

Comment:
I think this can be misleading as we `urlparse` does not do url validation.

Maybe we can be more specific on what we are trying to achieve here which is to update correctly the path.

So something like `def append_flush_path(url) -> str` which would extend the `url` path to have `/v2/flush`.

What do you think of the following?

```python
def append_flush_path(url) -> str:
    parts = list(urlsplit(url))
    # parts[2] corresponds to the url path
    # https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit
    parts[2] += "/v2/flush"
    return urlunsplit(parts)
```

and then we'd use like this:

```python
try:
    url = append_flush_path(cgw_url)
except ValueError:
    logger.error("CGW_URL is not a valid url")
    return
```

Reply:
On a second thought maybe just appending the path to the string-url would be enough. The main reason is that neither `urlparse` or `urlsplit` validate the `url`. Given that we can build the url with the following:

```python
url = cgw_url.rstrip('/') + "/v2/flush"
```